app/src/main/java/org/simple/clinic/recentpatient/RecentPatientsScreen.kt

```diff
@@ -82,8 +86,7 @@ class RecentPatientsScreen(context: Context, attrs: AttributeSet) : LinearLayout
         ))
   }
 
-  fun updateRecentPatients(allItemTypes: List<RecentPatientScreenItemTypes<out ViewHolder>>) {
-    allItemTypes.forEach { it.uiEvents = adapterEvents }
-    groupAdapter.update(allItemTypes)
+  fun updateRecentPatients(allItemTypes: List<RecentPatientItem>) {
+    recentAdapter.submitList(allItemTypes)
```

Comment:
I understand the function name describes it but can you help me understand this `submitList` ?

Reply:
It's a function in `ListAdapter`. It takes in a new list and calculates the diff and display the list in the `RecyclerView`. You can read more about it [here](https://developer.android.com/reference/androidx/recyclerview/widget/ListAdapter?authuser=1#submitList(java.util.List%3CT%3E)).